src/components/MARCTable/MARCTableRowContainer.js

```diff
@@ -13,38 +11,41 @@ export const MARCTableRowContainer = ({
   onAddNewRow,
   onRemoveRow,
   onMoveRow,
-  onDataChange,
-  intl,
 }) => {
-  const renderRow = (field, i) => {
-    const containsSubsequentLines = !isEmpty(field.subfields);
+  const renderRow = (data, i) => {
+    const containsSubsequentLines = data.field?.subfields?.length > 1;
+    const name = `profile.mappingDetails.marcMappingDetails[${i}]`;
 
     return (
       <div
-        data-test-marc-table-row={field.order}
+        data-test-marc-table-row={data.order}
         key={i}
         className={css.tableRowContainer}
       >
         <MARCTableRow
-          field={field}
+          name={name}
+          order={data.order}
+          action={data.action}
+          subaction={data.field?.subfields?.[0].subaction}
           columnWidths={columnWidths}
           isFirst={i === 0}
           isLast={i === (fields.length - 1)}
           onAddNewRow={onAddNewRow}
           onRemoveRow={onRemoveRow}
           onMoveRow={onMoveRow}
-          onDataChange={onDataChange}
-          intl={intl}
+          subfieldIndex={0}
         />
         {containsSubsequentLines &&
-          field.subfields.map(subfield => (
+          data.field.subfields.map((subfield, idx) => idx !== 0 && (
             <div>
```

Comment:
Since this is a `map`, should we have some sort of `key` attribute in this case?